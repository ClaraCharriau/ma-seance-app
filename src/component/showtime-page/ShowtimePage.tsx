import { Link, useLoaderData } from 'react-router-dom';
import { Showtime } from '../../models/Showtime';
import { ShowtimeDetails } from '../common/showtime-details/ShowtimeDetails';
import style from './ShowtimePage.module.css';
import config from '../../config/config.helper';

const ShowtimePage = () => {
    const showtime = useLoaderData() as Showtime;
    const { schedule, movie, theater } = showtime;
    const TMDB_PATH = config.tmdbImgPath.medium;

    return (
        <main className={style.showtimeMain}>
            <h2 className={style.pageTitle}>Séance à venir</h2>
            <section className={style.showtimePageSection}>
                <Link to={`/movies/${movie.id}/day-1`} state={{ movie: movie }} className={style.movieDesktopImg}>
                    <img src={TMDB_PATH + movie.posterLink} alt={`affiche du film ${movie.title}`} />
                </Link>
                <div className={style.rightColumn}>
                    <ShowtimeDetails screeningDate={schedule} movie={movie} theater={theater} isBooked={true} />
                    <button className={style.bookingButton}>
                        Réserver ma séance
                        <svg width="29" height="24" viewBox="0 0 29 24" fill="none" xmlns="http://www.w3.org/2000/svg">
                            <path
                                d="M7.23989 10.5462L19.4352 6.37522L21.8084 13.3139L9.61303 17.4849L7.23989 10.5462ZM23.9072 8.72093C24.3135 9.90906 25.4998 10.6093 26.6988 10.4453L27.901 13.9604C28.184 14.7878 27.7426 15.688 26.9152 15.971L7.20293 22.7128C6.37551 22.9958 5.47539 22.5545 5.19241 21.7271L3.99019 18.212C5.0384 17.6074 5.54743 16.3273 5.14108 15.1392C4.73472 13.9511 3.54842 13.2509 2.3495 13.4148L1.14728 9.8997C0.864297 9.07229 1.30562 8.17216 2.13304 7.88918L21.8453 1.1473C22.6727 0.864318 23.5729 1.30565 23.8559 2.13306L25.0581 5.64817C24.0099 6.25278 23.5008 7.53281 23.9072 8.72093ZM21.4214 4.80051C21.1459 3.99488 20.2694 3.56516 19.4638 3.8407L5.66519 8.56001C4.85956 8.83555 4.42984 9.71199 4.70538 10.5176L7.62685 19.0596C7.90239 19.8653 8.77883 20.295 9.58446 20.0194L23.3831 15.3001C24.1887 15.0246 24.6184 14.1481 24.3429 13.3425L21.4214 4.80051Z"
                                stroke="black"
                            />
                        </svg>
                    </button>
                    <button className={style.cancelButton}>Supprimer la séance</button>
                </div>
            </section>
        </main>
    );
};

export default ShowtimePage;
