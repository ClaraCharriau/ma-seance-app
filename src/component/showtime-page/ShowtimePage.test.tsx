import { render } from '@testing-library/react';
import ShowtimePage from './ShowtimePage';
import { BrowserRouter } from 'react-router-dom';
import mockShowtime from '../../mocks/showtimes/showtimes.json';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLoaderData: () => {
        return mockShowtime;
    }
}));

describe('showtime page component tests', () => {
    it('should render page', () => {
        // When
        const component = render(
            <BrowserRouter>
                <ShowtimePage />
            </BrowserRouter>
        );

        // Then
        expect(component.baseElement).toMatchSnapshot();
    });
});
