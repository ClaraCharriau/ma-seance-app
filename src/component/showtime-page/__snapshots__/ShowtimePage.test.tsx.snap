// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`showtime page component tests should render page 1`] = `
<body>
  <div>
    <main
      class="showtimeMain"
    >
      <h2
        class="pageTitle"
      >
        Séance passée
      </h2>
      <section
        class="showtimePageSection"
      >
        <a
          class="movieDesktopImg"
          href="/movies/cff01b1b-a943-4bcf-a396-afd80417150a/day-1"
        >
          <img
            alt="affiche du film Dune"
            src="https://image.tmdb.org/t/p/w500/qpyaW4xUPeIiYA5ckg5zAZFHvsb.jpg"
          />
        </a>
        <div
          class="rightColumn"
        >
          <div
            class="showtimeWrapper"
          >
            <div
              class="movieAndDateWrapper"
            >
              <div
                class="movieBox"
              >
                <a
                  href="/movies/cff01b1b-a943-4bcf-a396-afd80417150a/day-1"
                >
                  <img
                    alt="affiche du film Dune"
                    class="movieImg"
                    src="https://image.tmdb.org/t/p/w500/qpyaW4xUPeIiYA5ckg5zAZFHvsb.jpg"
                  />
                </a>
                <div>
                  <a
                    class="movieTitle"
                    href="/movies/cff01b1b-a943-4bcf-a396-afd80417150a/day-1"
                  >
                    Dune
                  </a>
                  <p
                    class="movieDetails"
                  >
                    2h46
                     ● 
                    2024
                  </p>
                </div>
              </div>
              <hr
                class="horizontalLineFirst"
              />
              <div
                class="dateBox"
              >
                <div>
                  <p
                    class="date"
                  >
                    <svg
                      fill="none"
                      height="17"
                      viewBox="0 0 16 17"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M2.78883 1H13.5218C14.5098 1 15.3107 1.82474 15.3107 2.84211V13.8947C15.3107 14.9121 14.5098 15.7368 13.5218 15.7368H2.78883C1.80089 15.7368 1 14.9121 1 13.8947V2.84211C1 1.82474 1.80089 1 2.78883 1Z"
                        fill-rule="evenodd"
                        stroke="#FFFDEF"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                      <path
                        d="M1 4.6842H15.3107"
                        stroke="#FFFDEF"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                    Date
                  </p>
                  <p
                    class="screeningDate"
                  >
                    dimanche
                     
                    17
                     
                    mars
                  </p>
                </div>
                <div>
                  <p
                    class="hourly"
                  >
                    <svg
                      fill="none"
                      height="14"
                      viewBox="0 0 14 14"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M7 14C5.61553 14 4.26216 13.5895 3.11101 12.8203C1.95987 12.0511 1.06266 10.9579 0.532846 9.67879C0.00303299 8.3997 -0.13559 6.99224 0.134506 5.63437C0.404603 4.2765 1.07129 3.02922 2.05026 2.05026C3.02922 1.07129 4.2765 0.404603 5.63437 0.134506C6.99224 -0.13559 8.3997 0.00303299 9.67879 0.532846C10.9579 1.06266 12.0511 1.95987 12.8203 3.11101C13.5895 4.26216 14 5.61553 14 7C14 8.85652 13.2625 10.637 11.9497 11.9497C10.637 13.2625 8.85652 14 7 14ZM7 1.07693C5.82853 1.07693 4.68336 1.42431 3.70932 2.07514C2.73527 2.72598 1.9761 3.65104 1.52779 4.73334C1.07949 5.81564 0.962193 7.00657 1.19074 8.15554C1.41928 9.3045 1.9834 10.3599 2.81175 11.1882C3.64011 12.0166 4.6955 12.5807 5.84447 12.8093C6.99343 13.0378 8.18436 12.9205 9.26667 12.4722C10.349 12.0239 11.274 11.2647 11.9249 10.2907C12.5757 9.31664 12.9231 8.17148 12.9231 7C12.9231 5.4291 12.299 3.92255 11.1882 2.81175C10.0775 1.70096 8.5709 1.07693 7 1.07693Z"
                        fill="#FFFDEF"
                      />
                      <path
                        d="M9.96192 10.5C9.89106 10.5004 9.82081 10.4868 9.7552 10.4601C9.6896 10.4333 9.62993 10.3938 9.57961 10.3439L6.61808 7.38233C6.56817 7.33202 6.52869 7.27235 6.50189 7.20674C6.4751 7.14114 6.46151 7.07089 6.46192 7.00002V2.69233C6.46192 2.54952 6.51865 2.41256 6.61963 2.31158C6.72062 2.2106 6.85758 2.15387 7.00038 2.15387C7.14319 2.15387 7.28015 2.2106 7.38113 2.31158C7.48212 2.41256 7.53885 2.54952 7.53885 2.69233V6.77925L10.3442 9.57925C10.3947 9.62931 10.4348 9.68886 10.4621 9.75448C10.4894 9.8201 10.5035 9.89048 10.5035 9.96156C10.5035 10.0326 10.4894 10.103 10.4621 10.1686C10.4348 10.2343 10.3947 10.2938 10.3442 10.3439C10.2939 10.3938 10.2342 10.4333 10.1686 10.4601C10.103 10.4868 10.0328 10.5004 9.96192 10.5Z"
                        fill="#FFFDEF"
                      />
                    </svg>
                    Horaire
                  </p>
                  <p
                    class="screeningHourly"
                  >
                    11:45
                  </p>
                </div>
              </div>
            </div>
            <hr
              class="horizontalLineSecond"
            />
            <div
              class="theaterBox"
            >
              <img
                alt="photo de UGC Ciné Cité Strasbourg Etoile"
                src="/assets/img/theaters/ugc-etoile-strasbourg.jpg"
              />
              <div
                class="theaterDetails"
              >
                <div>
                  <p
                    class="theaterName"
                  >
                    UGC Ciné Cité Strasbourg Etoile
                  </p>
                  <p
                    class="theaterAddress"
                  >
                    25, avenue du Rhin 67100 Strasbourg
                  </p>
                </div>
                <a
                  class="mapBtn"
                  href="https://maps.google.com/?q=UGC Ciné Cité Strasbourg Etoile 25, avenue du Rhin 67100 Strasbourg"
                  rel="noreferrer"
                  target="_blank"
                >
                  <svg
                    fill="none"
                    height="32"
                    viewBox="0 0 32 32"
                    width="32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M31 15V31L26 26L21 31L16 26L11 31L6 26L1 31V8L6 3L11 8L16 3"
                      stroke="#FFF1A7"
                      stroke-linejoin="round"
                      stroke-miterlimit="10"
                    />
                    <path
                      d="M31 7C31 3.686 28.314 1 25 1C21.686 1 19 3.686 19 7C19 12 25 17 25 17C25 17 31 12 31 7Z"
                      stroke="#FFF1A7"
                      stroke-linejoin="round"
                      stroke-miterlimit="10"
                    />
                    <path
                      d="M25 9C26.1046 9 27 8.10457 27 7C27 5.89543 26.1046 5 25 5C23.8954 5 23 5.89543 23 7C23 8.10457 23.8954 9 25 9Z"
                      stroke="#FFF1A7"
                      stroke-linejoin="round"
                      stroke-miterlimit="10"
                    />
                  </svg>
                  Y aller !
                </a>
              </div>
            </div>
          </div>
          <a
            class="bookingButton"
            href="https://www.ugc.fr"
            rel="noreferrer"
            target="_blank"
          >
            Réserver ma séance
            <svg
              fill="none"
              height="24"
              viewBox="0 0 29 24"
              width="29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.23989 10.5462L19.4352 6.37522L21.8084 13.3139L9.61303 17.4849L7.23989 10.5462ZM23.9072 8.72093C24.3135 9.90906 25.4998 10.6093 26.6988 10.4453L27.901 13.9604C28.184 14.7878 27.7426 15.688 26.9152 15.971L7.20293 22.7128C6.37551 22.9958 5.47539 22.5545 5.19241 21.7271L3.99019 18.212C5.0384 17.6074 5.54743 16.3273 5.14108 15.1392C4.73472 13.9511 3.54842 13.2509 2.3495 13.4148L1.14728 9.8997C0.864297 9.07229 1.30562 8.17216 2.13304 7.88918L21.8453 1.1473C22.6727 0.864318 23.5729 1.30565 23.8559 2.13306L25.0581 5.64817C24.0099 6.25278 23.5008 7.53281 23.9072 8.72093ZM21.4214 4.80051C21.1459 3.99488 20.2694 3.56516 19.4638 3.8407L5.66519 8.56001C4.85956 8.83555 4.42984 9.71199 4.70538 10.5176L7.62685 19.0596C7.90239 19.8653 8.77883 20.295 9.58446 20.0194L23.3831 15.3001C24.1887 15.0246 24.6184 14.1481 24.3429 13.3425L21.4214 4.80051Z"
                stroke="black"
              />
            </svg>
          </a>
          <button
            class="cancelButton"
          >
            Supprimer la séance
          </button>
        </div>
      </section>
    </main>
  </div>
  <div
    class="ReactModalPortal"
  />
</body>
`;
